src/DbPostgreSQLAdapter/Connection.cpp

```diff
@@ -0,0 +1,52 @@
+#include "stdafx.h"
+#include "Connection.h"
+
+#include "ConnectionConfiguration.h"
+#include "Database.h"
+
+#include "libpq-fe.h"
+
+namespace {
+	std::unique_ptr<PGresult, void(*)(PGresult*)> makePGResult(PGresult* result)
+	{
+		return std::unique_ptr<PGresult, void(*)(PGresult*)>(result, PQclear);
+	}
+}
+
+namespace systelab::db::postgresql {
+
+	Connection::Connection()
+	{}
+
+	std::unique_ptr<IDatabase> Connection::loadDatabase(IConnectionConfiguration& configuration)
+	{
+		PGconn* connection;
+		const std::string host = configuration.getParameter("host");
+		const std::string port = configuration.getParameter("port");
+		const std::string user = configuration.getParameter("user");
+		const std::string password = configuration.getParameter("password");
+		std::optional<std::string> dbName;
+		if (configuration.hasParameter("database"))
+		{
+			dbName = configuration.getParameter("database");
+		}
+
+		connection = PQsetdbLogin(host.data(), port.data(), nullptr, nullptr, (dbName ? dbName->data() : nullptr), user.data(), password.data());
```

Comment:
I would go with `.c_str()`instead of `.data()`. Since some time ago they are supposed to work the same, but there is still a minor difference (according to doc) being that the former only returns const char* while the later has also a non-const case.
Back in the day the former was null-terminated and the later wasn't, but in theory they behave the same in today's standard.

Reply:
I wasn't aware of this difference. I'll change it